docs/source/azure/examples/vmwithpublicip.rst

```diff
@@ -0,0 +1,13 @@
+Find Virtual Machines with Public IP address
+============================================
+
+.. code-block:: yaml
+
+     policies:
+       - name: vms-with-public-ip
+         resource: azure.vm
+         filters:
+          - type: network-interface
+            key: 'properties.ipConfigurations[].properties.publicIPAddress.id'
+            op: eq

```

Comment:
eq is the default, so a little redundant to document for an example. also not-null handling is a shortcut before the equality comparison.